<a href="https://colab.research.google.com/github/divya-project1/Intro-to-MI-Divyabharathiec029/blob/master/Placement_recuriment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Placement Predication ML-Project 2

# Libraries imported
*pandas


In [1]:
import pandas as pd


# contents of data file

Data file contains datasets of placement activity in some college. parameters that was used by organization are:

*number of students

*gender

*percentage in school

*board type

*choice of interest

*percentage in pu

*board type

*percentage in degree

*percentage in mba

*specialization in mba

*work experience

*pretest

*status

*salary if placed


In [2]:
dataset = pd.read_csv("/content/Placement_Data_Full_Class .csv")

In [3]:
dataset

sl_no gender  ssc_p    ssc_b  ...  specialisation  mba_p      status    salary
0        1      M  67.00   Others  ...          Mkt&HR  58.80      Placed  270000.0
1        2      M  79.33  Central  ...         Mkt&Fin  66.28      Placed  200000.0
2        3      M  65.00  Central  ...         Mkt&Fin  57.80      Placed  250000.0
3        4      M  56.00  Central  ...          Mkt&HR  59.43  Not Placed       NaN
4        5      M  85.80  Central  ...         Mkt&Fin  55.50      Placed  425000.0
..     ...    ...    ...      ...  ...             ...    ...         ...       ...
210    211      M  80.60   Others  ...         Mkt&Fin  74.49      Placed  400000.0
211    212      M  58.00   Others  ...         Mkt&Fin  53.62      Placed  275000.0
212    213      M  67.00   Others  ...         Mkt&Fin  69.72      Placed  295000.0
213    214      F  74.00   Others  ...          Mkt&HR  60.23      Placed  204000.0
214    215      M  62.00  Central  ...          Mkt&HR  60.22  Not Placed       NaN

[215 rows x 15 columns]

In [4]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 215 entries, 0 to 214
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   sl_no           215 non-null    int64  
 1   gender          215 non-null    object 
 2   ssc_p           215 non-null    float64
 3   ssc_b           215 non-null    object 
 4   hsc_p           215 non-null    float64
 5   hsc_b           215 non-null    object 
 6   hsc_s           215 non-null    object 
 7   degree_p        215 non-null    float64
 8   degree_t        215 non-null    object 
 9   workex          215 non-null    object 
 10  etest_p         215 non-null    float64
 11  specialisation  215 non-null    object 
 12  mba_p           215 non-null    float64
 13  status          215 non-null    object 
 14  salary          148 non-null    float64
dtypes: float64(6), int64(1), object(8)
memory usage: 25.3+ KB


In [5]:
dataset['status'].unique()

array(['Placed', 'Not Placed'], dtype=object)

In [6]:
y = dataset.iloc[:, -1].values
x = dataset.drop(['salary'],axis=1)


In [7]:
x

sl_no gender  ssc_p    ssc_b  ...  etest_p specialisation  mba_p      status
0        1      M  67.00   Others  ...     55.0         Mkt&HR  58.80      Placed
1        2      M  79.33  Central  ...     86.5        Mkt&Fin  66.28      Placed
2        3      M  65.00  Central  ...     75.0        Mkt&Fin  57.80      Placed
3        4      M  56.00  Central  ...     66.0         Mkt&HR  59.43  Not Placed
4        5      M  85.80  Central  ...     96.8        Mkt&Fin  55.50      Placed
..     ...    ...    ...      ...  ...      ...            ...    ...         ...
210    211      M  80.60   Others  ...     91.0        Mkt&Fin  74.49      Placed
211    212      M  58.00   Others  ...     74.0        Mkt&Fin  53.62      Placed
212    213      M  67.00   Others  ...     59.0        Mkt&Fin  69.72      Placed
213    214      F  74.00   Others  ...     70.0         Mkt&HR  60.23      Placed
214    215      M  62.00  Central  ...     89.0         Mkt&HR  60.22  Not Placed

[215 rows x 14 columns]

In [8]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 215 entries, 0 to 214
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   sl_no           215 non-null    int64  
 1   gender          215 non-null    object 
 2   ssc_p           215 non-null    float64
 3   ssc_b           215 non-null    object 
 4   hsc_p           215 non-null    float64
 5   hsc_b           215 non-null    object 
 6   hsc_s           215 non-null    object 
 7   degree_p        215 non-null    float64
 8   degree_t        215 non-null    object 
 9   workex          215 non-null    object 
 10  etest_p         215 non-null    float64
 11  specialisation  215 non-null    object 
 12  mba_p           215 non-null    float64
 13  status          215 non-null    object 
 14  salary          148 non-null    float64
dtypes: float64(6), int64(1), object(8)
memory usage: 25.3+ KB


In [9]:
dataset.describe()

sl_no       ssc_p  ...       mba_p         salary
count  215.000000  215.000000  ...  215.000000     148.000000
mean   108.000000   67.303395  ...   62.278186  288655.405405
std     62.209324   10.827205  ...    5.833385   93457.452420
min      1.000000   40.890000  ...   51.210000  200000.000000
25%     54.500000   60.600000  ...   57.945000  240000.000000
50%    108.000000   67.000000  ...   62.000000  265000.000000
75%    161.500000   75.700000  ...   66.255000  300000.000000
max    215.000000   89.400000  ...   77.890000  940000.000000

[8 rows x 7 columns]

In [10]:
x = dataset.iloc[: ,0:-1].values
y = dataset.iloc[:, -2].values

In [11]:
x

array([[1, 'M', 67.0, ..., 'Mkt&HR', 58.8, 'Placed'],
       [2, 'M', 79.33, ..., 'Mkt&Fin', 66.28, 'Placed'],
       [3, 'M', 65.0, ..., 'Mkt&Fin', 57.8, 'Placed'],
       ...,
       [213, 'M', 67.0, ..., 'Mkt&Fin', 69.72, 'Placed'],
       [214, 'F', 74.0, ..., 'Mkt&HR', 60.23, 'Placed'],
       [215, 'M', 62.0, ..., 'Mkt&HR', 60.22, 'Not Placed']], dtype=object)

In [12]:
y

array(['Placed', 'Placed', 'Placed', 'Not Placed', 'Placed', 'Not Placed',
       'Not Placed', 'Placed', 'Placed', 'Not Placed', 'Placed', 'Placed',
       'Not Placed', 'Placed', 'Not Placed', 'Placed', 'Placed',
       'Not Placed', 'Not Placed', 'Placed', 'Placed', 'Placed', 'Placed',
       'Placed', 'Placed', 'Not Placed', 'Placed', 'Placed', 'Placed',
       'Not Placed', 'Placed', 'Not Placed', 'Placed', 'Placed',
       'Not Placed', 'Placed', 'Not Placed', 'Placed', 'Placed', 'Placed',
       'Placed', 'Not Placed', 'Not Placed', 'Placed', 'Placed',
       'Not Placed', 'Not Placed', 'Placed', 'Placed', 'Not Placed',
       'Placed', 'Not Placed', 'Not Placed', 'Placed', 'Placed', 'Placed',
       'Placed', 'Placed', 'Placed', 'Placed', 'Placed', 'Placed',
       'Placed', 'Not Placed', 'Placed', 'Not Placed', 'Placed', 'Placed',
       'Not Placed', 'Placed', 'Placed', 'Placed', 'Placed', 'Placed',
       'Placed', 'Not Placed', 'Placed', 'Placed', 'Placed', 'Not Placed',
  

**separating the features and labels into two different dataframes.**

In [13]:
X = dataset.iloc[:,0:-1].values       
Y = dataset.iloc[:,-3].values

In [14]:
from sklearn.preprocessing import LabelEncoder

In [15]:
le = LabelEncoder()

In [16]:
x[:,1] = le.fit_transform(x[:,1])

In [17]:
from sklearn.compose import ColumnTransformer
from sklearn.preprocessing import OneHotEncoder
import numpy as np

In [18]:
ct = ColumnTransformer(transformers=[('encoder',OneHotEncoder(), [1,3,5,6,8,9,11,13])], remainder='passthrough')

In [19]:
x = np.array(ct.fit_transform(x))

In [20]:
x

array([[0.0, 1.0, 0.0, ..., 58.0, 55.0, 58.8],
       [0.0, 1.0, 1.0, ..., 77.48, 86.5, 66.28],
       [0.0, 1.0, 1.0, ..., 64.0, 75.0, 57.8],
       ...,
       [0.0, 1.0, 0.0, ..., 73.0, 59.0, 69.72],
       [1.0, 0.0, 0.0, ..., 58.0, 70.0, 60.23],
       [0.0, 1.0, 1.0, ..., 53.0, 89.0, 60.22]], dtype=object)

In [21]:
temp = pd.DataFrame(data = x)
temp

0  1  2  3  4  5  6  7  8   ... 15 16 17   18     19     20     21    22     23
0    0  1  0  1  0  1  0  1  0  ...  1  0  1    1     67     91     58    55   58.8
1    0  1  1  0  0  1  0  0  1  ...  0  0  1    2  79.33  78.33  77.48  86.5  66.28
2    0  1  1  0  1  0  1  0  0  ...  0  0  1    3     65     68     64    75   57.8
3    0  1  1  0  1  0  0  0  1  ...  1  1  0    4     56     52     52    66  59.43
4    0  1  1  0  1  0  0  1  0  ...  0  0  1    5   85.8   73.6   73.3  96.8   55.5
..  .. .. .. .. .. .. .. .. ..  ... .. .. ..  ...    ...    ...    ...   ...    ...
210  0  1  0  1  0  1  0  1  0  ...  0  0  1  211   80.6     82   77.6    91  74.49
211  0  1  0  1  0  1  0  0  1  ...  0  0  1  212     58     60     72    74  53.62
212  0  1  0  1  0  1  0  1  0  ...  0  0  1  213     67     67     73    59  69.72
213  1  0  0  1  0  1  0  1  0  ...  1  0  1  214     74     66     58    70  60.23
214  0  1  1  0  0  1  0  0  1  ...  1  1  0  215     62     58     53    89  60.22

[215 rows x 24 columns]

In [22]:
from sklearn.preprocessing import LabelEncoder

In [23]:
label_status = LabelEncoder()

In [24]:
dataset['status'] = label_status.fit_transform(dataset['status'])

In [25]:
dataset['status'].value_counts()

1    148
0     67
Name: status, dtype: int64

# Split into Test and Train data:

In [26]:
from sklearn.model_selection import train_test_split

In [27]:
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2)

# Scaling_data:


**Standardising Data**

*Standardizing the data means that it will transform the data so that its distribution will have a mean of 0 and a standard deviation of 1. It’s important to standardize your data in order to equalize the range of the data.

In [28]:
from sklearn.preprocessing import StandardScaler

In [29]:
sc = StandardScaler()

In [30]:
x_train = sc.fit_transform(x_train)
x_test = sc.transform(x_test)


# Classification Model

*Logistic Regression 

*KNN 

*Decision Tree 

*Random Forest

*linear SVC 

*Kernel SVC


In [31]:
from sklearn.linear_model import LogisticRegression
from sklearn.neighbors import KNeighborsClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.svm import SVC


In [32]:
l_cla =  LogisticRegression()
k_cla = KNeighborsClassifier()
d_cla = DecisionTreeClassifier()
r_cla = RandomForestClassifier()
s_cla = SVC(kernel = 'linear')
ks_cla = SVC(kernel = 'rbf')

In [33]:
l_cla.fit(x_train,y_train)
k_cla.fit(x_train,y_train)
d_cla.fit(x_train,y_train)
r_cla.fit(x_train,y_train)
s_cla.fit(x_train,y_train)
ks_cla.fit(x_train,y_train)


SVC(C=1.0, break_ties=False, cache_size=200, class_weight=None, coef0=0.0,
    decision_function_shape='ovr', degree=3, gamma='scale', kernel='rbf',
    max_iter=-1, probability=False, random_state=None, shrinking=True,
    tol=0.001, verbose=False)

In [34]:
l_pred = l_cla.predict(x_test)
k_pred = k_cla.predict(x_test)
d_pred = d_cla.predict(x_test)
r_pred = r_cla.predict(x_test)
s_pred = s_cla.predict(x_test)
ks_pred = ks_cla.predict(x_test)

In [35]:
from sklearn.metrics import confusion_matrix
l_c = confusion_matrix(y_test, l_pred)
k_c = confusion_matrix(y_test, k_pred)
d_c = confusion_matrix(y_test, d_pred)
r_c = confusion_matrix(y_test, r_pred)
s_c = confusion_matrix(y_test, s_pred)
ks_c = confusion_matrix(y_test, ks_pred)

In [36]:
from sklearn.metrics import accuracy_score

In [37]:
l_a = accuracy_score(y_test, l_pred)
k_a = accuracy_score(y_test, k_pred)
d_a = accuracy_score(y_test, d_pred)
r_a = accuracy_score(y_test, r_pred)
s_a = accuracy_score(y_test, s_pred)
ks_a = accuracy_score(y_test, ks_pred)


In [38]:
print('Logistic Regression: ' + str(l_a) + '\nKNN: ' + str(k_a) + '\nDecision Tree: ' + str(d_a) + '\nRandom Forest: ' + str(r_a) + '\nLinear SVC: ' + str(s_a) + '\nKernel SVC: ' + str(l_a))


Logistic Regression: 1.0
KNN: 0.9534883720930233
Decision Tree: 1.0
Random Forest: 1.0
Linear SVC: 1.0
Kernel SVC: 1.0


# Results:

After comparing with other algorithms with used algorithms gives following results:

Logistic Regression: 1.0

KNN: 0.9534883720930233

Decision Tree: 1.0

Random Forest: 1.0

Linear SVC: 1.0

Kernel SVC: 1.0

# Conclusion

According to given dataset,the student will be placed based on average of all percentage secured in school,pu,degree of above 60% and also pretest conducted by oraganization



Thank you

Divya bharathi.M